tests/lms/services/lis_result_sourcedid_test.py

```diff
@@ -8,6 +8,42 @@
 from lms import values
 
 
+class TestLISResultFetchStudentsByAssignment:
+    def test_it_retrieves_matching_records(
+        self, svc, lis_result_sourcedid, lis_result_sourcedid_info, lti_user, db_session
+    ):
+        h_user = values.HUser(
+            authority="TEST_AUTHORITY",
+            username="teststudent",
+            display_name="Another Test",
+        )
+        lti_user = values.LTIUser(
+            "TEST_USER_ID_2", "TEST_OAUTH_CONSUMER_KEY", "TEST_ROLES"
+        )
+        _add_lis_result_sourcedid(
+            lis_result_sourcedid_info, h_user, lti_user, db_session
+        )
+
+        students = svc.fetch_students_by_assignment(
+            oauth_consumer_key=lti_user.oauth_consumer_key,
+            context_id=lis_result_sourcedid_info.context_id,
+            resource_link_id=lis_result_sourcedid_info.resource_link_id,
+        )
+
+        assert len(students) == 2
+
+    def test_it_returns_empty_list_if_no_matching_records(
+        self, svc, lti_user, lis_result_sourcedid_info
+    ):
+        students = svc.fetch_students_by_assignment(
+            oauth_consumer_key=lti_user.oauth_consumer_key,
+            context_id=lis_result_sourcedid_info.context_id,
+            resource_link_id=lis_result_sourcedid_info.resource_link_id,
+        )
+
+        assert len(students) == 0
```

Comment:
```suggestion
        assert not students
```

(PEP8. This probably triggers a pylint warning)

Reply:
Fixed